### PR TITLE
Added presentation of sdf messages

### LINKED ISSUE #490: Robot importer, missing output from libsdf
Some of URDF fails to import, unfornuatelly reported error in URDF importer is useless:
image
It shows output from parsing SDF, but no from conversion process. 

# How to reproduce
Simply try to import `/opt/ros/humble/share/moveit_resources_panda_description/urdf/panda.urdf`

# Expected ouput:
```
Warning [parser_urdf.cc:2774] Error Code 19: Msg: urdf2sdf: link[panda_link0] has no <inertial> block defined. Please ensure this link has a valid mass to prevent any missing links or joints in the resulting SDF model.
Error Code 19: Msg: urdf2sdf: [1] child joints ignored.
Error Code 19: Msg: urdf2sdf: [1] child links ignored.
Warning [parser_urdf.cc:2777] Error Code 19: Msg: urdf2sdf: link[panda_link0] is not modeled in sdf.
```

### PR DESCRIPTION
**Pull request by michalpelka**
Fixes #490.
Added presentation of sdf messages:
Screenshot from 2023-09-04 16-02-04

### CODE REVIEW

File: Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp [hunk 1]
```diff
@@ -93,13 +94,25 @@ namespace ROS2::UrdfParser
...
             // Get any captured sdf::Error messages
         }
+        // escape console's color codes
+        // Define a regex pattern to match ANSI escape sequences
 
-        std::string parseMessages = parseStringStream.str();
-        parseResult.m_parseMessages = AZStd::string(parseMessages.c_str(), parseMessages.size());
+        // The pattern matches the following ANSI escape sequences:
+
+        // escape console's color codes
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp:106]: Seems like this comment is here twice and there's redundant whitespace / newlines

**Review by adamdbrw — APPROVED** [review]

**Review by jhanca-robotecai — APPROVED** [review]
The code works as expected, just minor comments

**Review by pawelbudziszewski — APPROVED** [review]

File: Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp [hunk 2]
```diff
@@ -93,13 +94,17 @@ namespace ROS2::UrdfParser
...
         }
+        const auto & parseMessages = parseStringStream.str();
 
-        std::string parseMessages = parseStringStream.str();
-        parseResult.m_parseMessages = AZStd::string(parseMessages.c_str(), parseMessages.size());
+        // regular exprsion to escape console's color codes
+        const AZStd::regex escapeColor("\x1B\\[[0-9;]*[A-Za-z]");
+
+        parseResult.m_parseMessages =  AZStd::regex_replace(AZStd::string(parseMessages.c_str(), parseMessages.size()),escapeColor, "");
+        AZ_Printf("ROS2", "SDF Stream: %s", parseResult.m_parseMessages.c_str());
```
Comment (lumberyard-employee-dm) [inline Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp:107]: There is no need to print the messages here as it gets printed in the added logic in the RobotImporterWidget.cpp OpenUrdf function.

The entire purpose of returning the parse messages as part of a result structure is to allow the **caller** to decide to print out the message, not function implementer.

Comment (michalpelka) [reply Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp:107]: Ok - I kept it to have this output present when URDF is loaded with ImportBus.
I've opened an issue to adjust the robot importer bus to follow your recommendations https://github.com/o3de/o3de-extras/issues/500.

**Pull request merged**